app/renderer/components/preferences/payment/enabledContent.js

```diff
@@ -0,0 +1,366 @@
+const settings = require('../../../../../js/constants/settings')
+const aboutActions = require('../../../../../js/about/aboutActions')
+
+// TODO: report when funds are too low
+// TODO: support non-USD currency
+class EnabledContent extends ImmutableComponent {
+  walletButton () {
+    const ledgerData = this.props.ledgerData
+    const buttonText = ledgerData.get('created')
+      ? 'addFundsTitle'
+      : (ledgerData.get('creating') ? 'creatingWallet' : 'createWallet')
+    const onButtonClick = ledgerData.get('created')
+      ? this.props.showOverlay.bind(this, 'addFunds')
+      : (ledgerData.get('creating') ? () => {} : this.createWallet())
+
+    return <Button
+      testId={buttonText}
+      test2Id={'addFunds'}
+      l10nId={buttonText}
+      className={css(commonStyles.buttonPrimary, styles.addFunds)}
+      onClick={onButtonClick.bind(this)}
+      disabled={ledgerData.get('creating')}
+    />
+  }
+
+  paymentHistoryButton () {
+    const ledgerData = this.props.ledgerData
+    const walletCreated = ledgerData.get('created') && !ledgerData.get('creating')
+    const walletTransactions = ledgerData.get('transactions')
+    const walletHasTransactions = walletTransactions && walletTransactions.size
+    const nextReconcileDateValue = this.nextReconcileDate()
+    let buttonText
+
+    if (!walletCreated || !nextReconcileDateValue) {
+      return null
+    } else if (!walletHasTransactions) {
+      buttonText = 'noPaymentHistory'
+      const now = new Date().getTime()
+      const timestamp = this.props.ledgerData.get('reconcileStamp')
+      if (timestamp <= now) {
+        buttonText = (timestamp <= (now - (24 * 60 * 60 * 1000)))
+          ? 'noPaymentOverDueHistory' : 'noPaymentDueHistory'
+      }
+    } else {
+      buttonText = 'viewPaymentHistory'
+    }
+
+    const l10nDataArgs = {
+      reconcileDate: nextReconcileDateValue
+    }
+
+    const onButtonClick = this.props.showOverlay.bind(this, 'paymentHistory')
+
+    return <Button
+      testId='paymentHistoryButton'
+      className={css(styles.paymentHistoryButton)}
+      l10nId={buttonText}
+      l10nArgs={l10nDataArgs}
+      onClick={onButtonClick.bind(this)}
+      disabled={ledgerData.get('creating')}
+    />
+  }
+
+  ledgerDataErrorText () {
+    const ledgerData = this.props.ledgerData
+    const ledgerError = ledgerData.get('error')
+
+    if (!ledgerError) {
+      return null
+    }
+
+    // 'error' here is a chromium webRequest error as returned by request.js
+    const errorCode = ledgerError.get('error').get('errorCode')
+    return l10nErrorText(errorCode)
+  }
+
+  walletStatus () {
+    const ledgerData = this.props.ledgerData
+    let status = {}
+
+    if (ledgerData.get('error')) {
+      status.id = 'statusOnError'
+    } else if (ledgerData.get('created')) {
+      const transactions = ledgerData.get('transactions')
+      const pendingFunds = Number(ledgerData.get('unconfirmed') || 0)
+
```

Comment:
This entire method (walletStatus) could benefit from moving over to ledgerUtil 😄  (it would need to be reworked to take ledgerData as input). That would also make mocking it easier